test/OpenTelemetry.Instrumentation.AspNetCore.Tests/ExceptionFilterTests.cs

```diff
@@ -0,0 +1,185 @@
+// <copyright file="ExceptionFilterTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Newtonsoft.Json;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Instrumentation.AspNetCore.Implementation;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+
+#if NETCOREAPP3_1
+using TestApp.AspNetCore._3._1;
+using TestApp.AspNetCore._3._1.Filters;
+#endif
+#if NET6_0
+using TestApp.AspNetCore._6._0;
+using TestApp.AspNetCore._6._0.Filters;
+#endif
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore.Tests
+{
+    public sealed class ExceptionFilterTests
+        : IClassFixture<WebApplicationFactory<Startup>>, IDisposable
+    {
+        private readonly WebApplicationFactory<Startup> factory;
+        private TracerProvider tracerProvider = null;
+
+        public ExceptionFilterTests(WebApplicationFactory<Startup> factory)
+        {
+            this.factory = factory;
+        }
+
+        [Fact]
+        public async Task ShouldExportExceptionWithNoExceptionFilters()
+        {
+            var exportedItems = new List<Activity>();
+
+            // Arrange
+            using (var client = this.factory
+                .WithWebHostBuilder(builder => builder.ConfigureTestServices(
+                    (s) => ConfigureTestServices(s, 0, ref exportedItems)))
+                .CreateClient())
+            {
+                // Act
+                var response = await client.GetAsync("/api/error");
+
+                WaitForActivityExport(exportedItems, 1);
+            }
+
+            // Assert
+            AssertException(exportedItems);
+        }
+
+        [Fact]
+        public async Task ShouldExportActivityWithOneExceptionFilter()
```

Comment:
Could move this test to `BasicTests.cs` to avoid have to maintain a copy of the helper methods `ValidateAspNetCoreActivity` and `WaitForActivityExport`.

Reply:
I removed the ExceptionFilterTests.cs from the project and moved both of them to BasicTests to avoid having to maintain a copy of `AssertException` on both of those